src/EscHandler.tsx

```diff
@@ -0,0 +1,33 @@
+import { useModals } from '@mattjennings/react-modal-stack'
+import React, { useEffect } from 'react'
+
+const useEscape = (onEscape: () => void) => {
+  useEffect(() => {
+    const handleEsc = (event: KeyboardEvent) => {
+      if (event.code === 'Escape' || event.keyCode === 27) {
+        onEscape()
+      }
+    }
+    window.addEventListener('keydown', handleEsc)
+
+    return () => {
+      window.removeEventListener('keydown', handleEsc)
+    }
+  })
+}
+
+export function EscHandler({ onClose }: { onClose?: () => any }) {
+  const { closeModal } = useModals()
+
+  function handleClose() {
+    if (onClose) {
+      onClose()
+    } else if (closeModal) {
+      closeModal()
+    }
+  }
+
+  useEscape(handleClose)
+
+  return <></>
+}
```

Comment:
Is there a need for both a component and a hook? I think it would be fine to add the effect for the key listener in the Modal component itself.

```ts
 useEffect(() => {
    if (allowEscKey) {
       const handleEsc = (event: KeyboardEvent) => {
         if (event.code === 'Escape' || event.keyCode === 27) {
           handleClose()
         }
       }
       window.addEventListener('keydown', handleEsc)

       return () => {
         window.removeEventListener('keydown', handleEsc)
       }
     }
   }, [allowEscKey])
 ```

Reply:
Totally, both ways would work, matter of taste. Adjusted accordingly.

Reply:
Awesome, looks good to me!